renter/renterutil/strategies.go

```diff
@@ -1306,7 +1307,7 @@ func (ssd SerialSectorDeleter) DeleteSectors(ctx context.Context, db MetaDB, sec
 		// TODO: respect ctx
 		err = h.DeleteSectors(roots)
 		ssd.Hosts.release(hostKey)
-		if err != nil {
+		if err != nil && !errors.Is(err, contractmanager.ErrSectorNotFound) {
```

Comment:
I don't think `errors.Is` works here, since the pointers won't match.

I'm also not sure whether this error should be handled here, or inside `DeleteSectors`. What do you think?

Reply:
Since `RPCError` implements "Is" interface, I think it works:

https://github.com/lukechampine/us/blob/dec6cf4b278e8f10a7b0dddd04a077fb0791d7f7/renterhost/session.go#L49-L52

`err` must be an `*RPCError`, mustn't it?

> I'm also not sure whether this error should be handled here, or inside DeleteSectors.

I think `proro.Session` is a relatively lower-level API and callers of `Session.DeleteSectors` would want to distinguish if the sectors are removed or don't already exist. On the other hand, callers of `SectorDeleter.DeleteSectors` might not mind if some sectors are already removed, and are interested in if the blob is removed from the hosts and DB.

So, I think it'd be better to handle the error in `SerialSectorDeleter.DeleteSectors` and `ParallelSectorDeleter`.

Reply:
oh yeah, I forgot that you implemented `Is` -- nice!

I agree, `Session` is low level and should report exactly what the host says.